opensearch/src/main/java/org/opensearch/sql/opensearch/executor/protector/OpenSearchExecutionProtector.java

```diff
@@ -116,6 +134,70 @@ public PhysicalPlan visitDedupe(DedupeOperator node, Object context) {
         node.getConsecutive());
   }
 
+  @Override
+  public PhysicalPlan visitLookup(LookupOperator node, Object context) {
+    return new LookupOperator(
+        visitInput(node.getInput(), context),
+        node.getIndexName(),
+        node.getMatchFieldMap(),
+        node.getAppendOnly(),
+        node.getCopyFieldMap(),
+        lookup());
+  }
+
+  private BiFunction<String, Map<String, Object>, Map<String, Object>> lookup() {
+
+    if (openSearchClient == null) {
+      throw new RuntimeException(
+          "Can not perform lookup because openSearchClient was null. This is likely a bug.");
+    }
+
+    return (indexName, inputMap) -> {
+      Map<String, Object> matchMap = (Map<String, Object>) inputMap.get("_match");
+      Set<String> copySet = (Set<String>) inputMap.get("_copy");
+
+      BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
+
+      for (Map.Entry<String, Object> f : matchMap.entrySet()) {
+        BoolQueryBuilder orQueryBuilder = new BoolQueryBuilder();
+
+        // Todo: Search with term and a match query? Or terms only?
```

Comment:
Not sure about this. Can you share an example query in which term and match both are required?

Reply:
When the lookup field in the lookup index is a "keyword" field (e.g. not analyzed) we need a term query. If the field is an analyzed field (full text field) we need a match query. Because we don't know the mapping of the field we just do both. Its way cheaper than first check the mapping.